src/wifi.rs

```diff
@@ -537,7 +551,7 @@ impl EspWifi {
             let mut shared = self.waitable.state.lock();
 
             shared.status = status.clone();
-            shared.operating = status.is_operating();
+            shared.operating = status.0.is_operating();
```

Comment:
Do you have some better suggestion for handling this? From what I can tell we only care about `operating` in `Station`-parts of the code?